lib/types.ts

```diff
@@ -6,17 +6,19 @@ export type DeepPartial<T> = {
     : DeepPartial<T[P]>;
 };
 
-export type GeneratorFnOptions<T, I> = {
+export type GeneratorFnOptions<T, I, C> = {
   sequence: number;
   afterBuild: (fn: HookFn<T>) => any;
-  onCreate: (fn: CreateFn<T>) => any;
+  afterCreate: (fn: AfterCreateFn<C>) => any;
+  onCreate: (fn: OnCreateFn<T, C>) => any;
   params: DeepPartial<T>;
   associations: Partial<T>;
   transientParams: Partial<I>;
 };
-export type GeneratorFn<T, I> = (opts: GeneratorFnOptions<T, I>) => T;
+export type GeneratorFn<T, I, C> = (opts: GeneratorFnOptions<T, I, C>) => T;
 export type HookFn<T> = (object: T) => any;
-export type CreateFn<T> = (object: T) => Promise<T>;
+export type OnCreateFn<T, C = T> = (object: T | C) => C | Promise<C>;
```

Comment:
After working with this for a while, this option seems to be fairly un-ergonomic because you have to cast `T | C` to `T` for Prisma.

Reply:
@casey-chow thanks for this feedback. You're right, I did not mean to commit this change. The signature should be `T => C | Promise<C>`. I had originally set this up as `T | C` when I was trying to get chaining to work but when I realized it would not be ergonomic to work with, like you mentioned, I switched to defining `afterCreate` separately from `onCreate`.

I just pushed an update that addresses this in 59ea435.